### PR TITLE
Rework entity props/config handling

### PR DESCRIPTION
**Pull request by allmarkedup**
Removes the concept of config data (and config data shadowing) from the Entity class, and tweaks how config/props are handled in Component instances. Also adds entity-specific toJSON methods to make sure all important info is serialised in the JSON representation.

### CODE REVIEW

File: packages/internals/support/src/collections/component-collection.test.js [hunk 1]
```diff
@@ -93,10 +93,6 @@ describe('ComponentCollection', function () {
       const collection = makeCollection();
       expect(collection.find('id', 'odie')).to.equal(collection[4]);
       expect(collection.find({
-        type: 'dog'
```
[the remark this answers is not in the document]
Comment (allmarkedup) [reply packages/internals/support/src/collections/component-collection.test.js]: I'm sure there was a reason... or more likely me just being heavy handed with the delete key! Will add it back in ;-)

File: packages/internals/support/src/entities/file.js [hunk 2]
```diff
@@ -202,8 +202,8 @@ class File extends Entity {
   }
```
[the remark this answers is not in the document]
Comment (allmarkedup) [reply packages/internals/support/src/entities/file.js]: Good spot, updated in latest commit

File: packages/internals/support/src/entities/variant.js [hunk 3]
```diff
@@ -12,18 +12,34 @@ const parser = new Parser5({locationInfo: true});
 const _templates = new WeakMap();
 const _componentId = new WeakMap();
 
+const reservedConfigProps = [
+  'opts',
+  'files',
+  'views',
+  'previews',
+  'scenarios',
+  'templates'
```
Comment (dkhuntrods) [inline packages/internals/support/src/entities/variant.js]: possibly also `component`? Or is this alterable?

Comment (allmarkedup) [reply packages/internals/support/src/entities/variant.js]: yup good point, will add it in

**Pull request merged**